labs/src/main/scala/org/scalalabs/basic/lab03/PatternMatchingExercise.scala

```diff
@@ -47,16 +52,26 @@ object PatternMatchingExercise {
    *    - anything else, the function result is "Some Scala class"
    */
   def matchOnInputType(in: Any): String = {
-    error("fix me")
+    in match{
+      case s:String => "A string with length " + s.length
+      case x:Int if (x>0) => "A positive integer"
+      case p:Person => "A person with name: " + p.name
+      case s:Seq[_] if s.length>10=> "Seq with more than 10 elements"
+      case s:Seq[_] if s.length>2=> "first: " + s(0) + ", second: " + s(1) + ", rest: " + List(s(2),s(3))  
```

Comment:
Shouldn't it be `", rest: " + s.slice(2,s.length)` ? It seems like if the list has more than 3 elements you won't display the entire tail.